src/App.vue

```diff
@@ -1,18 +1,26 @@
 <template lang="pug">
-v-app.app
-  girder-upload(v-if="folder", :dest="folder", :multiple="multiple")
-  v-layout(v-else, justify-center)
-    v-flex(xs12, sm10, md8, lg6)
-      girder-auth(
-          :register="true",
-          :oauth="true",
-          :forgot-password-url="forgotPasswordUrl")
+v-app.app(v-if="folder !== undefined")
+  v-dialog(v-model="loginDialog", width="50%", persistent)
+    girder-auth(
+        :register="true",
+        :oauth="true",
+        :forgot-password-url="forgotPasswordUrl")
+  v-dialog(v-model="uploader", width="70%")
+    girder-upload(v-if="uploader",
+        :dest="uploadDest",
+        multiple="multiple",
+        @done="refresh++")
```

Comment:
I don't especially like this, but I also don't see a way around it.  In order to force computed properties to re-compute without any of their dependencies having changed where the trigger is an event from Upload.vue, this seems like the best way to do it.  

On the plus side, if the client wanted to implement some kind of polling loop to keep the file browser up to date, this would make it trivial.

Reply:
https://forum.vuejs.org/t/force-computed-properties-to-recompute/2769

Not sure if this would also solve the problem, but this at least seems a little more simple and explicit (calling `this.$forceUpdate()` in one place seems more readable than having a mysterious `refresh` property that is mentioned in 5 places in the `.vue` file).

Reply:
@jeffbaumes I tried that first.  As far as I can tell, calling $forceUpdate() doesn't have any effect.  I tried calling it with `$refs` from the parent and calling a function in the child that then calls `this.$forceUpdate()` but the computed properties don't re-compute.

EDIT: Poked around and Evan confirms this: https://github.com/vuejs/vue/issues/7395#issuecomment-355721554

Looks like there is actually no way to do this without an explicit dependency update.

Reply:
Here is an officially sanctioned way to deal with this (which matches your approach well enough):

https://github.com/vuejs/vue/issues/214#issuecomment-57822806

However, I believe there is a more fundamental issue going on. It feels like using `asyncComputed` tarnishes the purity of computed properties. Computed properties are automagic and work because they are only supposed to depend on normal data properties. `asyncComputed` properties by definition depend on "other stuff", like server calls which actually hold the state, so we may find ourselves often bumping up against this impurity when using `asyncComputed`. It could be argued that we should not be using `asyncComputed`, but should have normal data properties for things like `count` and `rows`, and methods that mutate them with async code. That way, we would just call the appropriate method(s) when the upload completes, and things will reactively update as normal.

Reply:
I think you may be right about `asyncComputed`.  I do, however, like the organization (particularly the ease of forming a mental model of the dependency tree) it affords you.   @jbeezley do you have any insight about whether or not `asyncComputed` is appropriate in this case, since you've spent a lot more time with it than me?